lepton/starterkits/evo2-finetuning-and-prediction-with-bionemo2.ipynb

```diff
@@ -0,0 +1,1372 @@
+    "  random_lineage_dropout: 0.0\n",
+    "  include_sequence_id: false\n",
+    "  transcribe: \"back_transcribe\"\n",
+    "  force_uppercase: false\n",
+    "  indexed_dataset_dtype: \"uint8\"\n",
+    "  tokenizer_type: \"Byte-Level\"\n",
+    "  vocab_file: null\n",
+    "  vocab_size: null\n",
+    "  merges_file: null\n",
+    "  pretrained_tokenizer_model: null\n",
+    "  special_tokens: null\n",
+    "  fast_hf_tokenizer: true\n",
+    "  append_eod: true\n",
+    "  enforce_sample_length: null\n",
+    "  ftfy: false\n",
+    "  workers: 1\n",
+    "  preproc_concurrency: 100000\n",
+    "  chunksize: 25\n",
+    "  drop_empty_sequences: true\n",
+    "  nnn_filter: false  # If you split your fasta on NNN (in human these are contigs), then you should set this to true.\n",
+    "  seed: 12342  # Not relevant because we are not using random reverse complement or lineage dropout.\n",
+    "\"\"\"\n",
+    "with open(\"preprocess_config.yaml\", \"w\") as f:\n",
+    "    print(output_yaml, file=f)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "preprocess_evo2 --config preprocess_config.yaml"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!ls -lh {PREPROCESSED_DATA_DIR}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "After preprocessing, there should be a collection of bin/idx files created in the preprocessed_data directory."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### Configure the Training Dataset\n",
+    "To configure your training dataset, we create a yaml file specifying the paths for the training data we downloaded and preprocessed above."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "output_pfx = os.path.join(PREPROCESSED_DATA_DIR,\"chr20_21_22_uint8_distinct_byte-level\")\n",
+    "output_yaml = f\"\"\"\n",
+    "- dataset_prefix: {output_pfx}_train\n",
+    "  dataset_split: train\n",
+    "  dataset_weight: 1.0\n",
+    "- dataset_prefix: {output_pfx}_val\n",
+    "  dataset_split: validation\n",
+    "  dataset_weight: 1.0\n",
+    "- dataset_prefix: {output_pfx}_test\n",
+    "  dataset_split: test\n",
+    "  dataset_weight: 1.0\n",
+    "\"\"\"\n",
+    "with open(\"training_data_config.yaml\", \"w\") as f:\n",
+    "    print(output_yaml, file=f)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Specify or convert initial checkpoint\n",
+    "The main difference between pre-training and fine-tuning is whether or not you decide to start training the model with\n",
+    "weights from a prior training run. For this tutorial we want to finetune a `1b` checkpoint from HuggingFace that is known\n",
```

Comment:
Technically you can resume pre-training from an exsisting checkpoint with model weights (also continuous pre-training). Maybe something along the lines of "The main difference between pre-training and fine-tuning is that pre-training trains a model foundational knowledge from scratch, while fine-tuning refines an existing pre-trained model."